tests/conftest.py

```diff
@@ -35,6 +35,8 @@ def _setup():
 
 
 def pytest_addoption(parser):
+    os.environ['PYTHONWARNINGS'] = 'ignore:DEPRECATION::pip._internal.cli.base_command'  # for when not run within tox
```

Comment:
Can you provide some background on why this is needed?

Reply:
On python2.7 pip prints a warning that people should not use old python, which pollutes the stdout and causes issues when we parse the stdout to assert on content of it. This disables that warning. We know python2.7 is EOL, but we want to support it nonetheless, for us that warning is not needed.